swagger_spec_compatibility/rules/common.py

```diff
@@ -147,6 +175,8 @@ class BaseRule(with_metaclass(RuleRegistry)):
     error_level = None  # type: Level
     # Type of the rule associated
     rule_type = None  # type: RuleType
+    # Documentation link
+    documentation_link = None  # type: typing.Text
```

Comment:
Shouldn't some of these be Optional? E.g. you do have code in this PR handling the case that `documentation_link` is `None`.

All of these should probably use [ClassVar](https://mypy.readthedocs.io/en/latest/class_basics.html#class-attribute-annotations) since they're not supposed to be used from an instance.

Reply:
Fair point, updated BaseRule and RuleProtocol to expect ClassVar and made documentation_link optional

Reply:
Great, let's see how this goes. It might be better to use `typing_extensions.ClassVar` since it's not available in all versions of Python 3.5 .

Reply:
My bad on not reading all in the linked page. Going to do it right now :)